.devcontainer/android/Dockerfile

```diff
@@ -0,0 +1,58 @@
+ARG VARIANT="8.0-jammy"
+FROM mcr.microsoft.com/devcontainers/dotnet:${VARIANT}
+
+# Set up machine requirements to build the repo
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends \
+        cmake \
+        llvm \
+        clang \
+        build-essential \
+        python3 \
+        curl \
+        git \
+        lldb \
+        liblldb-dev \
+        libunwind8 \
+        libunwind8-dev \
+        gettext \
+        libicu-dev \
+        liblttng-ust-dev \
+        libssl-dev \
+        libkrb5-dev \
+        zlib1g-dev \
+        ninja-build \
+        openjdk-17-jdk \
+        pulseaudio
+
+SHELL ["/bin/bash", "-c"]
+
+ENV NDK_VER=r23c
```

Comment:
I wonder if we should use the existing container image that we use for building Android in CI as a base so we don't need to duplicate setting up the Android tooling: https://github.com/dotnet/runtime/blob/e994165856cf36e17ce72ec4c9b4cceea7132cfd/eng/pipelines/common/templates/pipeline-with-resources.yml#L42-L44

It is based on CBL-Mariner though so not sure if that works with Codespaces.

Reply:
I can try and see what happens.

Reply:
These Mariner containers are missing a lot of dependencies required to work as a Codespace. They don't have the necessary agents, node, shell configuration.

 Painful as it may be to have the duplication... it's a lot easier to duplicate setting up Android than to re-create everything needed for the container to function in a Codespace.

Reply:
ok, thanks for checking